openpype/hosts/maya/plugins/publish/validate_animation_out_set_related_node_ids.py

```diff
@@ -1,16 +1,18 @@
 import maya.cmds as cmds
-
 import pyblish.api
+
 import openpype.hosts.maya.api.action
 from openpype.hosts.maya.api import lib
 from openpype.pipeline.publish import (
+    OptionalPyblishPluginMixin,
+    PublishValidationError,
     RepairAction,
     ValidateContentsOrder,
-    PublishValidationError
 )
 
 
-class ValidateOutRelatedNodeIds(pyblish.api.InstancePlugin):
+class ValidateOutRelatedNodeIds(pyblish.api.InstancePlugin,
+                                OptionalPyblishPluginMixin):
```

Comment:
Unfortunately the changes made of only adding the mixin does not suffice. The `process` method should also be doing a `self.is_active(instance.data)` check to make sure it's actually skipped if the instance is toggled off. (Note that for ContextPlugin it is `self.is_active(context.data)`.

Reply:
> Also validators that ensure the correct behavior of the family for publishing should NEVER be optional.

True, but to really make them optional, there is another step and that is defining that in the Settings. My point is that even if we add the mixin, it doesn't really make the plugin optional, only the existance of the switch in the Settings does. I would even go that far later on as to unify inheritance of all validator plugin to one class that will wrap it together. Because then it wouldn't be *"confusing"* in the code as you wouldn't see `OptionalPyblishPluginMixin` in the signature of the validator class on non-optional plugins.